irondb/irondb.py

```diff
@@ -225,17 +302,18 @@ def find_nodes(self, query):
                 break
             except requests.exceptions.ConnectionError:
                 # on down nodes, try again on another node until "tries"
-                pass
+                log.debug("IRONdbFinder.find_nodes ConnectionError")
             except requests.exceptions.ConnectTimeout:
                 # on down nodes, try again on another node until "tries"
-                pass
+                log.debug("IRONdbFinder.find_nodes ConnectTimeout")
             except requests.exceptions.ReadTimeout:
                 # up node that simply timed out is a failure
+                log.debug("IRONdbFinder.find_nodes ReadTimeout")
                 break
-
+        log.debug("IRONdbFinder.find_nodes, result: %s" % json.dumps(names))
```

Comment:
The JSON serialization of the result to the logs can be very expensive, and runs regardless of whether it is printed or not here. This should only happen if the log level is set to debug.